pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -240,6 +256,69 @@ func (c *StaticResourceController) Name() string {
 	return "StaticResourceController"
 }
 
+func (c *StaticResourceController) RelatedObjects() ([]configv1.ObjectReference, error) {
+	if c.restMapper == nil {
+		return nil, errors.New("StaticResourceController.restMapper is nil")
+	}
+
+	if c.categoryExpander == nil {
+		return nil, errors.New("StaticResourceController.categoryExpander is nil")
+	}
+
+	// create lookup for resources in "all" alias
+	grs, _ := c.categoryExpander.Expand("all")
+	lookup := make(map[schema.GroupResource]struct{})
+	for _, gr := range grs {
+		lookup[gr] = struct{}{}
+	}
+
+	acc := make([]configv1.ObjectReference, 0)
+	errors := []error{}
+
+	for _, file := range c.files {
+		// parse static asset
+		objBytes, err := c.manifests(file)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		requiredObj, _, err := genericCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		// map gvk to gvr
+		gvk := requiredObj.GetObjectKind().GroupVersionKind()
+		mapping, err := c.restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		gvr := mapping.Resource
+		// filter out resources within "all" alias from result
+		if _, ok := lookup[gvr.GroupResource()]; ok {
+			continue
+		}
```

Comment:
You should skip members of `all` *only* for namespaced resources (i.e. `namespace != ""`)

Reply:
yes I assumed that since we parse static assets, it should be all namespaced, but I agree should be explicit, on it 👍🏽